File: solutions/python/triangle/1/triangle.py
def isTriangle(sides):
    return not any(x <= 0 for x in sides) and sides[0] + sides[1] >= sides[2] and sides[0] + sides[2] >= sides[1] and sides[1] + sides[2] >= sides[0]
    

def equilateral(sides):
    return isTriangle(sides) and sides[0] == sides[1] == sides[2]


def isosceles(sides):
    return isTriangle(sides) and (sides[0] == sides[1] or sides[0] == sides[2] or sides[1] == sides[2])


def scalene(sides):
    return isTriangle(sides) and (sides[0] != sides[1] and sides[0] != sides[2] and sides[1] != sides[2])

File: solutions/javascript/triangle/1/triangle.js
//
// This is only a SKELETON file for the 'Triangle' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class Triangle {
  constructor(...sides) {
    this.sides = sides;
  }

  get isEquilateral() {
    return this.isTriangle() && this.sides.every(n => n === this.sides[0]);
  }

  get isIsosceles() {
    return this.isTriangle() && (this.sides[0] === this.sides[1] || this.sides[0] === this.sides[2] || this.sides[2] === this.sides[1]);
  }

  get isScalene() {
    return this.isTriangle() && (new Set(this.sides)).size === this.sides.length;
  }

  isTriangle() {
    return this.sides.every(n => n > 0) && ((this.sides[0] + this.sides[1]) >= this.sides[2]) && ((this.sides[0] + this.sides[2]) >= this.sides[1]) && ((this.sides[2] + this.sides[1]) >= this.sides[0]);
  }
}
